dubbo-admin-backend/src/test/java/org/apache/dubbo/admin/controller/LoadBalanceControllerTest.java

```diff
@@ -0,0 +1,147 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+public class LoadBalanceControllerTest extends AbstractSpringIntegrationTest {
+
+    private final String env = "whatever";
+
+    @MockBean
+    private OverrideService overrideService;
+    @MockBean
+    private ProviderService providerService;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    public void createLoadbalance() throws IOException {
+        BalancingDTO balancingDTO = new BalancingDTO();
+        ResponseEntity<String> response;
+
+        // service and application are all blank
+        response = restTemplate.postForEntity(url("/api/{env}/rules/balancing"), balancingDTO, String.class, env);
+        assertFalse("should return a fail response, when service and application are all blank", (Boolean) objectMapper.readValue(response.getBody(), Map.class).get("success"));
+        // dubbo version is 2.6
+        balancingDTO.setApplication("test application");
+        balancingDTO.setService("test service");
+        when(providerService.findVersionInApplication("test application")).thenReturn("2.6");
+        response = restTemplate.postForEntity(url("/api/{env}/rules/balancing"), balancingDTO, String.class, env);
+        assertFalse("should return a fail response, when dubbo version is 2.6", (Boolean) objectMapper.readValue(response.getBody(), Map.class).get("success"));
+        // dubbo version is 2.7
+        when(providerService.findVersionInApplication("test application")).thenReturn("2.7");
+        response = restTemplate.postForEntity(url("/api/{env}/rules/balancing"), balancingDTO, String.class, env);
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
+        assertTrue(Boolean.valueOf(response.getBody()));
+    }
+
+    @Test
+    public void updateLoadbalance() throws IOException {
+        String id = "1";
+        BalancingDTO balancingDTO = new BalancingDTO();
+        URI uri;
+        ResponseEntity<String> response;
+        // unknown id
+        uri = restTemplate.getUriTemplateHandler().expand(url("/api/{env}/rules/balancing/{id}"), env, id);
+        response = restTemplate.exchange(uri, HttpMethod.PUT, new HttpEntity<>(balancingDTO), String.class);
+        assertFalse("should return a fail response, when id is null", (Boolean) objectMapper.readValue(response.getBody(), Map.class).get("success"));
+        // valid id
+        BalancingDTO balancing = mock(BalancingDTO.class);
+        when(overrideService.findBalance(id)).thenReturn(balancing);
+        uri = restTemplate.getUriTemplateHandler().expand(url("/api/{env}/rules/balancing/{id}"), env, id);
+        response = restTemplate.exchange(uri, HttpMethod.PUT, new HttpEntity<>(balancingDTO), String.class);
+        assertTrue(Boolean.valueOf(response.getBody()));
```

Comment:
Why not specify the return type as Boolean.class? Then there is no need to use ObjectMapper, you can use responseEntity.getBody()